### PR TITLE
Add documentation for advanced property migration

### PR DESCRIPTION
**Pull request by AndrewGaspar**
Rough draft - I'm sure there needs to be a link in some TOC somewhere.

### CODE REVIEW

File: windows-driver-docs-pr/network/driver-upgrades-advanced-property-keywords.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+---
+
+# Advanced Properties Migration
+
+Because [advanced properties](specifying-configuration-parameters-for-the-advanced-properties-page.md) contain user specified state, that user state must be preserved on driver upgrades and after Windows updates. This is to align with user expectations that their settings are permanent. As of Windows 10 Creator's Update (1703), user state will always be preserved for registry values that a driver specifies as being advanced keywords. This is a break from previous versions of Windows that would always re-configure the INF on-top of user state.
+
+Resetting user state after a driver or Windows update breaks user's expectations. That's why it's important to manage this state carefully. If a driver needs to reset an advanced property on upgrade, the driver should detect the first time a driver is started after upgrade and perform the reset in code, but it should only perform the reset exactly once and allow the user to continue to configure the driver to their liking.
```
Comment (jtippet) [inline windows-driver-docs-pr/network/driver-upgrades-advanced-property-keywords.md]: Don't mention "detect the first time after upgrade" unless you're prepared to explain how to detect that.  (And we really prefer that drivers not peek at the OS version.)  Better to just say that drivers should always clamp keywords to an allowed range, which is doubly important if the range might change in a different driver version.

Comment (jtippet) [inline windows-driver-docs-pr/network/driver-upgrades-advanced-property-keywords.md]: Maybe I'm old-fashioned, but... _the user_ is singular and _their_ is (to me, still) plural.

File: windows-driver-docs-pr/network/driver-upgrades-advanced-property-keywords.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+It is recommended to mark any advanced properties in the INF where the default may change in the future as "Optional". This gives the driver the freedom to re-interpret the default behavior in the future, while only falling back to the old default for users that explicitly configured the property.
```
Comment (jtippet) [inline windows-driver-docs-pr/network/driver-upgrades-advanced-property-keywords.md]: This point isn't super clear to me.  Can you rephrase or give an example?

**Comment by tedhudek** [comment]
@windows-driver-content I'm working on this PR out-of-band with Andrew

**Comment by tedhudek** [comment]
Thanks for the PR, @AndrewGaspar this update is now live at https://msdn.microsoft.com/en-us/windows/hardware/drivers/network/specifying-configuration-parameters-for-the-advanced-properties-page (note up top)
